flytekit/tools/script_mode.py

```diff
@@ -231,6 +235,12 @@ def list_imported_modules_as_files(source_path: str, modules: List[ModuleType])
             # in the site-packages or bin directory.
             pass
 
+        if any(
+            (__drive_safe_commonpath([site_package, mod_file]) == site_package for site_package in site_packages_set)
```

Comment:
Given the access pattern, I think we can remove `site_packages_set` and use `site_packages` directly:

```suggestion
            (__drive_safe_commonpath([site_package, mod_file]) == site_package for site_package in site_packages)
```